src/python/WMComponent/TaskArchiver/CleanCouchPoller.py

```diff
@@ -380,6 +396,16 @@ def deleteWorkflowFromWMBSAndDisk(self):
         deletableWorkflowsDAO = self.daoFactory(classname="Workflow.GetDeletableWorkflows")
         deletablewfs = deletableWorkflowsDAO.execute()
 
+        # For T0 subtract the workflows which are not having all their blocks deleted yet:
+        if not self.useReqMgrForCompletionCheck:
+            undeletedBlocksByWorkflowDAO = self.dbsDaoFactory(classname="CountUndeletedBlocksByWorkflow")
```

Comment:
I'd suggest to make it an object instance and initialize it in the `setup` method.

Reply:
I do not see a reason why to spawn these changes in many places, since the rule of thumb for the whole `CleanCouchPoller` module (at least as far as I notice) is:
 * create the DAO instance where (in the method) it is needed
 * execute the DAO in place (right after instantiating it)

I know in other modules  the practice is different. But we would not gain in terms of CPU or anything if we move that in the object instance. And this DAO is about to be used here and here only. While keeping the whole thing well grouped in one place seems to me more logical and easy to link which is what, and where it is used.

Reply:
It does take a few extra CPU cycles to try to instantiate it every cycle, but I have strong preference here.